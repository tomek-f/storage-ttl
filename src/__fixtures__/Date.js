global.Date = global.Date || {};
global.Date.now = jest.fn(() => 1513977789823);
