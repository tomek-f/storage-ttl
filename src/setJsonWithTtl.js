import isNumber from '@tomekf/number/dist/is';

const { stringify: toJson } = JSON;

const setJonWithTtl = storage => (name, value, ttlInMilisec = 0) => {
  if (!name || !isNumber(ttlInMilisec)) {
    throw new Error(!name ? '`name` arg needed' : '`ttlInMilisec` arg must be a number');
  }
  storage.setItem(name.toString(), toJson({ value, expiresAt: Date.now() + ttlInMilisec }));
};

export default setJonWithTtl;
