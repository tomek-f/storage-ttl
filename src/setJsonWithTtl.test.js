import './__fixtures__/Date';

import setJsonWithTtl from './setJsonWithTtl';

const { localStorage } = global;
const setJsonTtl = setJsonWithTtl(localStorage);

test('object snapshot', () => {
  setJsonTtl('yolo', { a: 1 }, 2000);

  expect(localStorage.getItem('yolo')).toMatchSnapshot();
});

test('undefined snapshot', () => {
  setJsonTtl('yolo', undefined, 3000);
  expect(localStorage.getItem('yolo')).toMatchSnapshot();
});

test('null snapshot', () => {
  setJsonTtl('yolo', null, 4000);
  expect(localStorage.getItem('yolo')).toMatchSnapshot();
});

test('number snapshot', () => {
  setJsonTtl('yolo', 1, 5000);
  expect(localStorage.getItem('yolo')).toMatchSnapshot();
});

test('string snapshot', () => {
  setJsonTtl('yolo', 'yolo', 5000);
  expect(localStorage.getItem('yolo')).toMatchSnapshot();
});

test('object snapshot minus ts', () => {
  setJsonTtl('yolo', { a: 1 }, -2000);
  expect(localStorage.getItem('yolo')).toMatchSnapshot();
});

test('object snapshot no ttlInMilisec', () => {
  setJsonTtl('yolo', { a: 1 });
  expect(localStorage.getItem('yolo')).toMatchSnapshot();
});

test('no name error', () => {
  expect(() => setJsonTtl()).toThrow('`name` arg needed');
});

test('bad ttlInMilisec error', () => {
  expect(() => setJsonTtl('a', 1, '1111')).toThrow('`ttlInMilisec` arg must be a number');
});
