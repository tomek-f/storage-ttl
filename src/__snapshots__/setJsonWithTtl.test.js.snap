// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`null snapshot 1`] = `"{\\"value\\":null,\\"expiresAt\\":1513977793823}"`;

exports[`number snapshot 1`] = `"{\\"value\\":1,\\"expiresAt\\":1513977794823}"`;

exports[`object snapshot 1`] = `"{\\"value\\":{\\"a\\":1},\\"expiresAt\\":1513977791823}"`;

exports[`object snapshot minus ts 1`] = `"{\\"value\\":{\\"a\\":1},\\"expiresAt\\":1513977787823}"`;

exports[`object snapshot no ttlInMilisec 1`] = `"{\\"value\\":{\\"a\\":1},\\"expiresAt\\":1513977789823}"`;

exports[`string snapshot 1`] = `"{\\"value\\":\\"yolo\\",\\"expiresAt\\":1513977794823}"`;

exports[`undefined snapshot 1`] = `"{\\"expiresAt\\":1513977792823}"`;
